server/src/main/java/org/apache/iotdb/db/metadata/MTree.java

```diff
@@ -800,6 +800,7 @@ private void findDevices(MNode node, String[] nodes, int idx, String parent, Has
       for (MNode child : node.getChildren().values()) {
         if(child.isLeaf()){
           res.add(parent + node.getName());
+          break;
```

Comment:
"break" is not correct here. The child of a device may also be device, e.g., we have timeseries root.a.s1 and root.a.b.s2, so both root.a and root.a.b are devices. (A device may consist of smaller devices).

Reply:
Fix it.